### PR TITLE
Virtual Host expose Rate Limit Policy

### PR DESCRIPTION
**Pull request by ccaraman**

### CODE REVIEW

**Comment by ccaraman** [comment]
@lyft/network-team

File: docs/configuration/http_conn_man/route_config/vhost.rst [hunk 1]
```diff
@@ -53,3 +54,7 @@ require_ssl
 :ref:`virtual_clusters <config_http_conn_man_route_table_vcluster>`
   *(optional, array)* A list of virtual clusters defined for this virtual host. Virtual clusters
   are used for additional statistics gathering.
+
+:ref:`rate_limits <config_http_conn_man_route_table_rate_limit_config>`
+  *(optional, array)* Specifies a set of rate limit configurations that could be applied to the
```
Comment (mattklein123) [inline docs/configuration/http_conn_man/route_config/vhost.rst]: "that will be applied" also please link to rate limit arch overview / other applicable docs

File: source/common/http/filter/ratelimit.cc [hunk 2]
```diff
@@ -25,16 +25,32 @@ FilterHeadersStatus Filter::decodeHeaders(HeaderMap& headers, bool) {
...
 
     std::vector<::RateLimit::Descriptor> descriptors;
-    for (const Router::RateLimitPolicyEntry& rate_limit :
+
+    // Get all applicable rate limit policy entries for the route.
+    for (const Router::RateLimitPolicyEntry& route_rate_limit :
          route->rateLimitPolicy().getApplicableRateLimit(config_->stage())) {
-      const std::string& route_key = rate_limit.routeKey();
+      const std::string& route_key = route_rate_limit.routeKey();
+      if (!route_key.empty() &&
```
Comment (mattklein123) [inline source/common/http/filter/ratelimit.cc]: You can create a small helper method which takes a list of policies, and then just call it twice (once for route and once for vhost) then you don't duplicate all of this code.

File: docs/configuration/http_conn_man/route_config/vhost.rst (hunk 1 above)
Comment (ccaraman) [inline docs/configuration/http_conn_man/route_config/vhost.rst]: Will update wording. Links to rate limit arch overview is included in the <..._rate_limit_config> doc..

**Pull request merged**